added partial solution

razvanrus2003 opened the pull request:
[#5](https://github.com/ilai-deutel/kibi/issues/5)
-added support for utf8 characters for save and find commands.
-highlight bugs for fullwidth characters

razvanrus2003 committed:
added partial solution

razvanrus2003 committed:
better highlight length getter (using unicode_width)

razvanrus2003 commented:
-fixed highlight bug by using unicode_width crate to determine String display width instead of the number of bits/characters

razvanrus2003 committed:
finnished coding style

razvanrus2003 commented:
- final solution

alexandruradovici reviewed (changes requested):
Please fix the build issues.

razvanrus2003 committed:
fixed warnings

alexandruradovici reviewed (approved):
Please send the pull request to upstream and put a link to it here.

razvanrus2003 commented:
https://github.com/ilai-deutel/kibi/pull/281
here

razvanrus2003 committed:
Lines optimization (#280)

* -2 lines

* modified macro, -2 lines

ilai-deutel committed:
add auzkok as contributor for syntax-highlighting (#278)

ilai-deutel committed:
[CI] Fail on Clippy error (#283)

ilai-deutel committed:
Merge branch 'master' into utf8_prompts

ilai-deutel committed:
[CI] Actually fail on Clippy error (#284)

dependabot[bot] committed:
Bump libc from 0.2.152 to 0.2.153 (#286)

Bumps [libc](https://github.com/rust-lang/libc) from 0.2.152 to 0.2.153.
- [Release notes](https://github.com/rust-lang/libc/releases)
- [Commits](https://github.com/rust-lang/libc/compare/0.2.152...0.2.153)

---
updated-dependencies:
- dependency-name: libc
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump tempfile from 3.9.0 to 3.10.0 (#289)

Bumps [tempfile](https://github.com/Stebalien/tempfile) from 3.9.0 to 3.10.0.
- [Changelog](https://github.com/Stebalien/tempfile/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Stebalien/tempfile/compare/v3.9.0...v3.10.0)

---
updated-dependencies:
- dependency-name: tempfile
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump tempfile from 3.9.0 to 3.10.0 in /fuzz (#288)

Bumps [tempfile](https://github.com/Stebalien/tempfile) from 3.9.0 to 3.10.0.
- [Changelog](https://github.com/Stebalien/tempfile/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Stebalien/tempfile/compare/v3.9.0...v3.10.0)

---
updated-dependencies:
- dependency-name: tempfile
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump tempfile from 3.10.0 to 3.10.1 in /fuzz (#290)

Bumps [tempfile](https://github.com/Stebalien/tempfile) from 3.10.0 to 3.10.1.
- [Changelog](https://github.com/Stebalien/tempfile/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Stebalien/tempfile/compare/v3.10.0...v3.10.1)

---
updated-dependencies:
- dependency-name: tempfile
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump tempfile from 3.10.0 to 3.10.1 (#291)

Bumps [tempfile](https://github.com/Stebalien/tempfile) from 3.10.0 to 3.10.1.
- [Changelog](https://github.com/Stebalien/tempfile/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Stebalien/tempfile/compare/v3.10.0...v3.10.1)

---
updated-dependencies:
- dependency-name: tempfile
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump unicode-width from 0.1.11 to 0.1.12 (#294)

Bumps [unicode-width](https://github.com/unicode-rs/unicode-width) from 0.1.11 to 0.1.12.
- [Commits](https://github.com/unicode-rs/unicode-width/compare/v0.1.11...v0.1.12)

---
updated-dependencies:
- dependency-name: unicode-width
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump winapi-util from 0.1.6 to 0.1.8 (#295)

Bumps [winapi-util](https://github.com/BurntSushi/winapi-util) from 0.1.6 to 0.1.8.
- [Commits](https://github.com/BurntSushi/winapi-util/compare/0.1.6...0.1.8)

---
updated-dependencies:
- dependency-name: winapi-util
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump libc from 0.2.153 to 0.2.155 (#297)

Bumps [libc](https://github.com/rust-lang/libc) from 0.2.153 to 0.2.155.
- [Release notes](https://github.com/rust-lang/libc/releases)
- [Commits](https://github.com/rust-lang/libc/compare/0.2.153...0.2.155)

---
updated-dependencies:
- dependency-name: libc
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump libc from 0.2.155 to 0.2.158 (#301)

dependabot[bot] committed:
Bump tempfile from 3.10.1 to 3.12.0 in /fuzz (#299)

dependabot[bot] committed:
Bump winapi-util from 0.1.8 to 0.1.9 (#302)

dependabot[bot] committed:
Bump libc from 0.2.158 to 0.2.159 (#303)

Bumps [libc](https://github.com/rust-lang/libc) from 0.2.158 to 0.2.159.
- [Release notes](https://github.com/rust-lang/libc/releases)
- [Changelog](https://github.com/rust-lang/libc/blob/0.2.159/CHANGELOG.md)
- [Commits](https://github.com/rust-lang/libc/compare/0.2.158...0.2.159)

---
updated-dependencies:
- dependency-name: libc
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump tempfile from 3.12.0 to 3.13.0 in /fuzz (#304)

Bumps [tempfile](https://github.com/Stebalien/tempfile) from 3.12.0 to 3.13.0.
- [Changelog](https://github.com/Stebalien/tempfile/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Stebalien/tempfile/compare/v3.12.0...v3.13.0)

---
updated-dependencies:
- dependency-name: tempfile
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ilai-deutel committed:
Bump MSRV to Rust 1.65.0 (#306)

dependabot[bot] committed:
Bump serial_test from 3.0.0 to 3.1.1 (#293)

Bumps [serial_test](https://github.com/palfrey/serial_test) from 3.0.0 to 3.1.1.
- [Release notes](https://github.com/palfrey/serial_test/releases)
- [Commits](https://github.com/palfrey/serial_test/compare/v3.0.0...v3.1.1)

---
updated-dependencies:
- dependency-name: serial_test
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ilai-deutel committed:
But MSRV to 1.66.0 (#307)

dependabot[bot] committed:
Bump unicode-width from 0.1.12 to 0.2.0 (#300)

Bumps [unicode-width](https://github.com/unicode-rs/unicode-width) from 0.1.12 to 0.2.0.
- [Commits](https://github.com/unicode-rs/unicode-width/commits)

---
updated-dependencies:
- dependency-name: unicode-width
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ilai-deutel committed:
[CI] Migrate WASI targets (#316)

See [announcement](https://blog.rust-lang.org/2024/04/09/updates-to-rusts-wasi-targets.html)

ilai-deutel committed:
[CI] Upgrade code coverage generation (#317)

dependabot[bot] committed:
Bump tempfile from 3.10.1 to 3.13.0 (#309)

Bumps [tempfile](https://github.com/Stebalien/tempfile) from 3.10.1 to 3.13.0.
- [Changelog](https://github.com/Stebalien/tempfile/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Stebalien/tempfile/compare/v3.10.1...v3.13.0)

---
updated-dependencies:
- dependency-name: tempfile
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump tempfile from 3.13.0 to 3.14.0 in /fuzz (#315)

Bumps [tempfile](https://github.com/Stebalien/tempfile) from 3.13.0 to 3.14.0.
- [Changelog](https://github.com/Stebalien/tempfile/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Stebalien/tempfile/compare/v3.13.0...v3.14.0)

---
updated-dependencies:
- dependency-name: tempfile
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump libfuzzer-sys from 0.4.7 to 0.4.8 in /fuzz (#314)

Bumps [libfuzzer-sys](https://github.com/rust-fuzz/libfuzzer) from 0.4.7 to 0.4.8.
- [Changelog](https://github.com/rust-fuzz/libfuzzer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/rust-fuzz/libfuzzer/compare/0.4.7...0.4.8)

---
updated-dependencies:
- dependency-name: libfuzzer-sys
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump libc from 0.2.159 to 0.2.162 (#313)

Bumps [libc](https://github.com/rust-lang/libc) from 0.2.159 to 0.2.162.
- [Release notes](https://github.com/rust-lang/libc/releases)
- [Changelog](https://github.com/rust-lang/libc/blob/0.2.162/CHANGELOG.md)
- [Commits](https://github.com/rust-lang/libc/compare/0.2.159...0.2.162)

---
updated-dependencies:
- dependency-name: libc
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump serial_test from 3.1.1 to 3.2.0 (#312)

Bumps [serial_test](https://github.com/palfrey/serial_test) from 3.1.1 to 3.2.0.
- [Release notes](https://github.com/palfrey/serial_test/releases)
- [Commits](https://github.com/palfrey/serial_test/compare/v3.1.1...v3.2.0)

---
updated-dependencies:
- dependency-name: serial_test
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

malbarbo committed:
Use alternate screen (#310)

ilai-deutel committed:
[CI] Upgrade action versions (#319)

allcontributors[bot] committed:
add Jan9103 as a contributor for code (#318)

* update README.md [skip ci]

* update .all-contributorsrc [skip ci]

---------

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>

ilai-deutel committed:
Update rusfmt config and fix associated clippy errors (#321)

ilai-deutel committed:
Update README.md - HelloGitHub claim

Signed-off-by: Ilaï Deutel <ilai-deutel@users.noreply.github.com>

allcontributors[bot] committed:
add joshka as a contributor for infra (#324)

* update README.md [skip ci]

* update .all-contributorsrc [skip ci]

---------

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>

Commit:
fix: crash when opening a new file (#287)

Bug Reproduction: rm foo.txt; kibi foo.txt

ilai-deutel committed:
[CI] Configure secret token for reviewdog (#332)

ilai-deutel committed:
[Clippy] Check in config, update CI (#333)

ilai-deutel committed:
Combine mod xdg with cfg(any) (LOC -= 1) (#335)

ilai-deutel committed:
[Clippy] Add nursery-level warnings, fix issues found (#334)

ilai-deutel committed:
Fix formatting and Clippy workflows outside of pull requests (#336)

joshka committed:
Replace Key::Page(PageKey) with Key::Page{Up,Down} (#331)

Saves 4 LOG

allcontributors[bot] committed:
add joshka as a contributor for code (#337)

* update README.md [skip ci]

* update .all-contributorsrc [skip ci]

---------

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>

joshka committed:
Add vscode settings for using nightly rustfmt (#329)

This makes vscode automatically use the correct formatting when saving

ilai-deutel committed:
[CI] Update security workflows, add cargo-deny, deprecate cargo-audit (#338)

ilai-deutel committed:
Fix config parsing with invalid durations (#340)

ilai-deutel committed:
Bump MSRV to 1.80 (#343)

ilai-deutel committed:
[Tests] Remove dependency on serial_test, use Mutex instead (#341)

This enables the use of sanitizers

ilai-deutel committed:
Fix markdownlint errors in README.md

ilai-deutel committed:
Delete .github/workflows/devskim.yml

Signed-off-by: Ilaï Deutel <ilai-deutel@users.noreply.github.com>

ilai-deutel committed:
Add weekly sanitizer run to CI (#342)

ilai-deutel committed:
[CI] Temporarily stop using mbrobbel/rustfmt-check

See https://github.com/mbrobbel/rustfmt-check/issues/1137

joshka committed:
Use module paths instead of conditional module includes for sys modules (#330)

Saves 1 LOC

joshka committed:
Replace count_loc.sh with xtask (#328)

* Replace count_loc.sh with xtask

To run the count_loc now, you can use `cargo xtask count-loc` instead of
`./count_loc.sh`.

See <https://github.com/matklad/cargo-xtask> for more info on
cargo-xtask.

Fixes: https://github.com/ilai-deutel/kibi/issues/323

* Only check formatting on main package, not xtask

There's a +nightly oddity where it returns an empty array twice instead of valid json for multiple projects, this causes the check action to fail

ilai-deutel committed:
[xtask] Add more info to Cargo.toml (#344)

ilai-deutel committed:
Fix lint config for main crate

ilai-deutel committed:
Remove obsolete TODO

Signed-off-by: Ilaï Deutel <ilai-deutel@users.noreply.github.com>

ilai-deutel committed:
Add more lints to check and clippy configuration (#345)

Commit:
feat: enable LTO (#346)

Signed-off-by: Alexander Zaitsev <zamazan4ik@tut.by>

allcontributors[bot] committed:
add zamazan4ik as a contributor for infra (#347)

BanuAdrian committed:
Adding test for home + end keys (#311)

* add test for pressing the home key

allcontributors[bot] committed:
add BanuAdrian as a contributor for test (#320)

* update README.md [skip ci]

* update .all-contributorsrc [skip ci]

---------

Signed-off-by: Ilaï Deutel <ilai-deutel@users.noreply.github.com>
Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>
Co-authored-by: Ilaï Deutel <ilai-deutel@users.noreply.github.com>

ilai-deutel committed:
Merge branch 'master' into utf8_prompts

Signed-off-by: Ilaï Deutel <ilai-deutel@users.noreply.github.com>
